plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -698,12 +698,15 @@ private Namespace toNamespace(String schemaName)
 
     private String toSchemaName(Namespace namespace)
     {
-        if (!nestedNamespaceEnabled && namespace.length() != 1) {
-            throw new TrinoException(NOT_SUPPORTED, "Nested namespace is not enabled for this catalog");
-        }
         if (this.parentNamespace.isEmpty()) {
+            if (!nestedNamespaceEnabled && namespace.length() != 1) {
```

Comment:
Added this commit as `TestIcebergUnityRestCatalogConnectorSmokeTest` was failing